sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RetryUtils.java

```diff
@@ -107,7 +107,7 @@
                             shouldRetryResult, minBackoffForInBackoffCallback, rxDocumentServiceRequest)
                             .delaySubscription(Duration.ofMillis(shouldRetryResult.backOffTime.toMillis()));
                 }
-            }).single();
+            });
```

Comment:
This will error on an empty Mono correct?

Could you explain on which retryPolicy code-path, retry policy returns an empty mono?

RetryPolicy is expected to never return empty Mono if it does we should discover that code path and fix that code path.